cmd/integration/harness.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+const execPathLabel = "executable-path"
+
+// TODO: this should be a util function
+// Contains tells whether a contains x.
+// if it does it returns it's index otherwise -1
+func Contains(a []string, x string) int {
+	for ind, n := range a {
+		if strings.Contains(n, x) {
+			return ind
+		}
+	}
+
+	return -1
+}
+
+// Harness fully encapsulates an active node server process
+// along with client connection and full api, created for rpc integration
+// tests and may be used for any other purpose.
+type Harness struct {
+	server *server
+}
+
+func NewHarnessDefaultServerConfig(args []string) (*Harness, error) {
+	// same as in suite's yaml file
+	// find executable path label in args
+	execPathInd := Contains(args, execPathLabel)
+	if execPathInd == -1 {
+		return nil, fmt.Errorf("could not find executable path in arguments")
+	}
+	// next will be exec path value
+	execPath := args[execPathInd+1]
+	// remove executable path label and value
+	args = append(args[:execPathInd], args[execPathInd+2:]...)
+	// set servers' configuration
+	cfg, errCfg := DefaultConfig(execPath)
+	if errCfg != nil {
+		return nil, fmt.Errorf("failed to build mock node: %v", errCfg)
+	}
+	return NewHarness(cfg, args)
+}
+
+// NewHarness creates and initializes a new instance of Harness.
+func NewHarness(cfg *ServerConfig, args []string) (*Harness, error) {
+	log.Info("Starting harness")
+	server, err := newServer(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	// kill node process in case it is already up
+	if isListening(cfg.rpcListen) {
```

Comment:
this will kill the harness. not the node process. is this expected behaviour?

Reply:
this kills any process listening on the harness port so the port will be available for the harness use.

Reply:
is this code needed? think about it, you are running a single exe in a container, how can two processes exist? 
this is legacy code from the actual harness built for testing, in case we run them multiple times a harness can still be running when running a new one. this behaviour won't happen in the docker

Reply:
removed in ac94cfaaaa15ee08f1ea691282076d8e3c81ef61